src/palette.c

```diff
@@ -578,10 +582,15 @@ static u8 UpdateFastPaletteFade(void)
     s8 b;
 
     if (!gPaletteFade.active)
-        return 0;
+        return PALETTE_FADE_STATUS_DONE;
 
-    if (IsSoftwarePaletteFadeFinishing())
-        return gPaletteFade.active;
+    if (IsSoftwarePaletteFadeFinishing()) {
```

Comment:
This change might cause the assembly to change. Check and make sure it does not.